vmm/modules/reset/reset.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2015-2022 Intel Corporation.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include "gcpu.h"
+#include "host_cpu.h"
+#include "kvm_workaround.h"
+
+#include "modules/io_monitor.h"
+#include "modules/ipc.h"
+
+#define CF9_BIT_SYS_RST  (1u << 1)
+#define CF9_BIT_RST_CPU  (1u << 2)
+#define CF9_BIT_FULL_RST (1u << 3)
+
+static void clear_vmx(guest_cpu_handle_t gcpu, void *vmx_off_cpu_cnt)
+{
+	guest_cpu_handle_t gcpu_next = gcpu;
+
+	do {
+		vmcs_clr_ptr(gcpu_next->vmcs);
+
+		gcpu_next = gcpu_next->next_same_host_cpu;
+	} while (gcpu_next != gcpu);
+
+	vmx_off();
+
+	if (vmx_off_cpu_cnt) {
+		(*((uint16_t *)vmx_off_cpu_cnt))++;
+		__STOP_HERE__;
```

Comment:
move __STOP_HERE__ out of "if"?

Reply:
here is to stop all 'APs', if move out, then there is no chance for 'BSP' to perform actual reset.

Reply:
then, please add a comment in line 32 that, vmx_off_cpu_cnt is NULL for BSP.

Reply:
done